kvstore/simplemap.go

```diff
@@ -0,0 +1,75 @@
+package kvstore
```

Comment:
For consistency with badger (and other key-value stores we'll add in the future), I think we should create a package called `simplemap` and have an associated `interface.go` file.

It's a NIT but trying to set a pattern for others to follow.

Reply:
I think badger is the pattern and simplemap is never likely to be used in prod. I am against making MapStore a submodule it needs to be accessable to the tries

Reply:
Going to push back against this. The way I see it is:

- `Simplemap` is an example (and the default) that won't be used in prod
- `Badger` is the first `prod` implementation
- There should be a share (base) interface

Happy to jump on a call and discuss more if you think it would help.